### DISCUSSION TITLE
Explicitly exclude files from tailwind `content` config option

### DISCUSSION THREAD

**Discussion by benjaminpreiss** [tailwindlabs/tailwindcss#12892] [Help] [answered]
I would like to exclude all `./src/**/*.tdc.ts` files from tailwind parsing.

My tailwind content config:

```ts
content: ['./src/**/*.{html,js,svelte,ts}']
```

Ideally, I'd like to specify this from my tailwind plugin or preset, so without changing the content config in the original config file.

Glad for any help on this!

**Comment by wongjn** [comment, marked as answer]
You can do in your Tailwind content config:

```js
content: [
  './src/**/*.{html,js,svelte,ts}',
  '!./src/**/*.tdc.ts',
],
```
But I assume this does not satisfy the stipulation of "without changing the content config in the original config file".

You can try adding it to the a Tailwind Plugin's default configuration, but I'm not sure if the merging logic works here:

```js
plugin(
  ( … ) => {
    …
  },
  {
    content: [
      '!./src/**/*.tdc.ts',
    ],
  },
);
```

And defining the negative file glob in a preset would not work either since it is replaced by the `content` in your Tailwind configuration.

**Reply by benjaminpreiss** [reply]
Cool, thanks a lot! Do you know whether tailwind follows import paths?

If e.g. a `.tdc.ts` file gets imported by a `.ts` file - will the classes in the `.tdc.ts` file be imported?

**Reply by wongjn** [reply]
Tailwind does not follow import paths.

**Reply by benjaminpreiss** [reply]
So - specifying the content option in a plugin is not working. It does however work as you said when manually adding this to the `content` option.

**Reply by heymartinadams** [reply]
Is there a way to exclude `node_modules`? I use a monorepo and share my `tailwind.config.js` across the entire monorepo, and also don’t use the `src` folder, so I need to be able to exclude all `node_modules`.

`'!./**/node_modules',`

**Reply by bitttttten** [reply]
any update on that @heymartinadams ? we're looking for a similar solution too, we have nested node_modules and tailwind is taking a lot of time to parse all our node_modules and it's hard to ignore them
